src/borg_import/main.py

```diff
@@ -1,17 +1,160 @@
+
+        Then you can define *import_something* like this in the class::
+
+            def import_something(self, args):
+                ...  # do something!
+        """
+
+
+class rsnapshotImporter(Importer):
+    name = 'rsnapshot'
+    description = 'import rsnapshot backups'
+
+    def populate_parser(self, parser):
+        parser.add_argument('--backup-set', help='Only consider given backup set (can be given multiple times).',
+                            action='append', dest='backup_sets')
+        parser.add_argument('rsnapshot_root', metavar='RSNAPSHOT_ROOT',
+                            help='Path to rsnapshot root directory', type=Path)
+        # TODO: support the full wealth of borg possibilities
+        parser.add_argument('repository', metavar='REPOSITORY', help='Borg repository', type=Path)
+        parser.set_defaults(function=self.import_rsnapshot)
+
+    def import_rsnapshot(self, args):
+        existing_archives = list_borg_archives(args)
+
+        import_path = args.rsnapshot_root / 'borg-import-dir'
+        import_journal = args.rsnapshot_root / 'borg-import-dir.snapshot'
+
+        if import_path.exists():
+            print('{} exists. Cannot continue.'.format(import_path))
+            return 1
+
+        for rsnapshot in get_snapshots(args.rsnapshot_root):
+            timestamp = rsnapshot['timestamp'].replace(microsecond=0)
+            snapshot_original_path = rsnapshot['path']
+            name = rsnapshot['name']
+            archive_name = args.prefix + name
+
+            if args.backup_sets and rsnapshot['backup_set'] not in args.backup_sets:
+                print('Skipping (backup set is not selected):', name)
+                continue
+
+            if archive_name in existing_archives:
+                print('Skipping (already exists in repository):', name)
+                continue
+
+            print('Importing {} (timestamp {}) '.format(name, timestamp), end='')
+            if archive_name != name:
+                print('as', archive_name)
+            else:
+                print()
+            log.debug('  Moving {} -> {}'.format(rsnapshot['path'], import_path))
+
+            with import_journal.open('w') as fd:
+                fd.write('Current snapshot: %s\n' % rsnapshot['name'])
+                fd.write('Original path: %s\n' % snapshot_original_path)
+
+            snapshot_original_path.rename(import_path)
+
+            try:
+                borg_import(args, archive_name, import_path, timestamp=timestamp)
+            finally:
+                log.debug('  Moving {} -> {}'.format(import_path, rsnapshot['path']))
+                import_path.rename(snapshot_original_path)
+                import_journal.unlink()
+
 
 def main():
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--rsnapshot-root", help="Path to rsnapshot root directory")
+    if not shutil.which('borg'):
+        print('Borg does not seem to be installed. Please install Borg first.')
```

Comment:
Maybe rather:

"The 'borg' command can't be found in the PATH. Please correctly install borgbackup first."